planning/behavior_path_sampling_planner_module/include/behavior_path_sampling_planner_module/constraints_checker.hpp

```diff
@@ -0,0 +1,114 @@
+  const sampler_common::Path &, const sampler_common::Constraints &)>;
+
+double checkSoftConstraints(
+  const sampler_common::Path & path, const sampler_common::Constraints & constraints,
+  const SoftConstraintsFunctionVector & soft_constraints, std::vector<double> & constraints_results)
+{
+  std::vector<double> constraints_evaluations(soft_constraints.size(), false);
+  double constraints_evaluation = 0.0;
+  for (const auto & f : soft_constraints) {
+    constraints_evaluation += f(path, constraints);
+  }
+  return constraints_evaluation;
+}
+
+bool checkHardConstraints(
+  const sampler_common::Path & path, const sampler_common::Constraints & constraints,
+  const HardConstraintsFunctionVector & hard_constraints, std::vector<double> & constraints_results)
+{
+  std::vector<double> constraints_evaluations(hard_constraints.size(), false);
+  bool constraints_evaluation = true;
+  for (const auto & f : hard_constraints) {
+    constraints_evaluation &= f(path, constraints);
+  }
+  return constraints_evaluation;
+}
+
+// template <typename T>
+// class ConstraintsChecker
+// {
+// private:
+//   /* data */
+// public:
+//   ConstraintsChecker(ConstraintsFunctionVector constraints_functions)
+//   : constraints_functions_(constraints_functions){};
+//   ~ConstraintsChecker(){};
+//   ConstraintsFunctionVector<T> constraints_functions_;
+
+//   T checkConstraints(
+//     const sampler_common::Path & path, const sampler_common::Constraints & constraints,
+//     std::vector<double> & constraints_results) = 0;
+
+//   template <>
+//   double checkConstraints(
+//     const sampler_common::Path & path, const sampler_common::Constraints & constraints,
+//     std::vector<double> & constraints_results)
+//   {
+//     std::vector<double> constraints_evaluations(constraints_functions_.size(), false);
+//     double constraints_evaluation = 0.0;
+//     for (const auto & f : constraints_functions_) {
+//       constraints_evaluation += f(path, constraints);
+//     }
+//     return constraints_evaluation;
+//   }
+
+//   template <>
+//   bool checkConstraints(
+//     const sampler_common::Path & path, const sampler_common::Constraints & constraints,
+//     std::vector<bool> & constraints_results)
+//   {
+//     std::vector<bool> constraints_evaluations(constraints_functions_.size(), false);
+//     bool all_constraints_passed = true;
+//     for (const auto & f : constraints_functions_) {
+//       all_constraints_passed &= f(path, constraints);
+//     }
+//     return all_constraints_passed;
+//   }
+// };
+
+// class HardConstraintsChecker : public ConstraintsChecker
+// {
+// private:
+//   /* data */
+// public:
+//   HardConstraintsChecker(/* args */){};
+//   ~HardConstraintsChecker(){};
+// };
+
+// class SoftConstraintsChecker : public ConstraintsChecker
+// {
+// private:
+//   /* data */
+// public:
+//   SoftConstraintsChecker(/* args */){};
+//   ~SoftConstraintsChecker(){};
+// };
+
```

Comment:
Are you planning on merging with these commented out?
If it is not used, might as well removed it from the code.

Reply:
Ah, my bad, I deleted it